tracer/src/Datadog.Trace/DuckTyping/DuckType.Methods.cs

```diff
@@ -1047,20 +1046,58 @@ internal static Type AddIlToDuckChain(LazyILGenerator il, Type genericType, Type
                     {
                         DuckTypeException.Throw($"CreateCache<{genericType}>.CreateFrom<{fromType}>() cannot be found!");
                     }
+
+                    il.Emit(OpCodes.Call, getProxyMethodInfo);
+                }
+                else if (genericType.IsGenericType && genericType.GetGenericTypeDefinition() == typeof(Nullable<>))
+                {
+                    // Support for Nullable<T>
+                    var argGenericType = genericType.GenericTypeArguments[0];
+                    var getProxyMethodInfo = typeof(CreateCache<>)
+                                        .MakeGenericType(argGenericType)
+                                        .GetMethod("Create");
+
+                    if (getProxyMethodInfo is null)
+                    {
+                        DuckTypeException.Throw($"CreateCache<{argGenericType}>.Create() cannot be found!");
+                    }
+
```

Comment:
Is this what the C# roughly looks like? Might be useful to add comments like these in the future so it's easier to parse at a glance? 🙂 

```suggestion

                    // T local;
                    // if (x is null)
                    // {
                    //     local = null;
                    // }
                    // else
                    // {
                    //     local = new T(CreateCache<T>.Create());
                    // }
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/4582/commits/f7c6d6f57bef843bf96c5902f4b43b8bbd11ad79